library/core/src/result.rs

```diff
@@ -2003,3 +2009,17 @@ impl<T, E, F: From<E>> ops::FromResidual<ops::Yeet<E>> for Result<T, F> {
 impl<T, E> ops::Residual<T> for Result<convert::Infallible, E> {
     type TryType = Result<T, E>;
 }
+
+#[cfg(kani)]
+#[unstable(feature="kani", issue="none")]
+mod verify {
+    use super::*;
+
+    #[kani::proof_for_contract(Result::unwrap_unchecked)]
+    pub fn check_unwrap_unchecked() {
+        let val: Result<u32, u64> = kani::any();
+        let ok_variant: Result<u32, u64> = Ok(0);
+        let copy = unsafe { ok_variant.unwrap_unchecked() };
+        assert_eq!(val, Result::Ok(copy));
```

Comment:
Since `val` is a nondeterministic `Result`, it's not necessary equal to `Result::Ok(0)` (it could also be `Result::Err`, or `Result::Ok(12)`, etc.
Instead, I'd write:

```
pub fn check_unwrap_unchecked() {
        let val: u32= kani::any();
        let ok_variant: Result<u32, u64> = Ok(val);
        let copy = unsafe { ok_variant.unwrap_unchecked() };
        assert_eq!(val, copy);
    }
```
which passes verification for me locally.

(I'd also add harnesses for other types besides `u32` and for `unwrap_err_unchecked`, but I assume you were planning on that).

Reply:
I am still curious why Kani couldn't report this issue usefully though...